### PR TITLE
Change default SERVICE_DOMAIN_ALGO to hdr_beg.

### PR DESCRIPTION
**Pull request by foleymic**
Address issue #359 

If published port is something other than 80 or 443, browsers will include the port in the Request's Header Host attribute.  If HAProxy's ACL is checking for `hdr(host)` to be some domain, it will not match.  Instead, we will use `hdr_beg(host)` to only look at the begging of the Host value for a match.

Also updated tests to use `hdr_beg` as the default `serviceDomainAlgo`

### CODE REVIEW

**Comment by vfarcic** [comment]
Can you also update the docs?

**Comment by foleymic** [comment]
Docs have been updated to reflect the new default value.  Sorry for missing that.  Let me know if you think I need to any text around why we're using hdr_beg.

File: docs/config.md [hunk 1]
```diff
@@ -37,7 +37,7 @@ PROXY_INSTANCE_NAME|The name of the proxy instance. Useful if multiple proxies a
...
 |SEPARATOR          |The character used to separate multiple values.<br>**Default value:** `,` (comma)|
-|SERVICE_DOMAIN_ALGO|The default algorithm applied to domain ACLs. It can be overwritten for a service through the `serviceDomainAlgo` parameter.<br>**Examples:**<br>`hdr(host)`: matches only if domain is the same as `serviceDomain`<br>`hdr_dom(host)`: matches the specified `serviceDomain` and any subdomain (a string either isolated or delimited by dots).<br>`req.ssl_sni`: matches Server Name TLS extension<br>**Default Value:** `hdr(host)`|
+|SERVICE_DOMAIN_ALGO|The default algorithm applied to domain ACLs. It can be overwritten for a service through the `serviceDomainAlgo` parameter.<br>**Examples:**<br>`hdr_beg(host)`: matches only if domain begins with `serviceDomain`<br>`hdr_dom(host)`: matches the specified `serviceDomain` and any subdomain (a string either isolated or delimited by dots).<br>`req.ssl_sni`: matches Server Name TLS extension<br>**Default Value:** `hdr_beg(host)`|
```
Comment (vfarcic) [inline docs/config.md]: I'd leave `hdr(host)` as the option as well. In other words, add `hdr_beg(host)` to the list and leave it as the default value.

File: docs/usage.md [hunk 2]
```diff
@@ -57,7 +57,7 @@ The following query parameters can be used only when `reqMode` is set to `http`
...
+|serviceDomainAlgo|Algorithm that should be applied to domain ACLs. Any ACL works only with one flag: `-i : ignore case during matching of all subsequent patterns`. If not set, the value of the environment variable `SERVICE_DOMAIN_ALGO` will be used instead. If defaults to `hdr_beg(host)`<br>**Examples:**<br>`hdr_beg(host)`: matches only if domain begins with `serviceDomain`<br>`hdr_dom(host)`: matches the specified `serviceDomain` and any subdomain (a string either isolated or delimited by dots). **Example:** if `hdr_dom(host)` contains `www.ecme.com` and `serviceDomain` equals `ecme.com` the rule will be passed.<br>`req.ssl_sni`: matches Server Name TLS extension|
```
Comment (vfarcic) [inline docs/usage.md]: The same comment as the one for `SERVICE_DOMAIN_ALGO`.

Comment (foleymic) [reply docs/usage.md]: Thanks @vfarcic.  I made the suggested changes to the docs.

**Comment by vfarcic** [comment]
Thank you 👍 

It's building and will be released in a few minutes.

**Pull request merged**